### PR TITLE
Hyperledger Solang annual report

### PR DESCRIPTION
**Pull request by salaheldinsoliman**

### CODE REVIEW

File: project-reports/2024/2024-annual-Hyperledger-Solang.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+# Project Adoption
+
+The maintainer team has identified the issue with Solang's slow adoption in 2023: the lack of developer tooling and advocacy. Although a lot of work has gone into making a full-on compiler for Solidity, we need to provide easy ways for Solidity developers to start using Solang. The maintainer team will work on tutorials and easy-to-use tools to supplement Solang.
+
+
+# Goals
+
+## Performance Against Prior Goals
+Prior goals for Solang mainly involved making Solang more attractive to Solidity developers as stated in the 2023 [Q4 plans](https://https://github.com/hyperledger/toc/blob/gh-pages/project-reports/2023/2023-Q4-Hyperledger-Solang.md#current-plans)
```
Comment (yacovm) [inline project-reports/2024/2024-annual-Hyperledger-Solang.md]: broken link?

File: project-reports/2024/2024-annual-Hyperledger-Solang.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+The 2024 forecast for Solang is showing growth despite the number of pull requests and reported issues going down, Here are the three reasons why:
+1- Stellar organization decided to fund Solang long term, to add Soroban as a compilation target.
+
+2- Work is ongoing in Hyperledger Solang Playground, a web IDE that will enable easy onboarding to the Solang ecosystem. Funding is provided by the Web3 Foundation. 
+
+3- Hyperledger Solang-Aqd (a smart contract CLI tool) was developed in the 2023 mentorship, by Tarek Naser, and it will be integrated into Solang Playground.
+
+# Maintainer Diversity
+- Another maintainer from a different organization has been added: Salaheldin Soliman. His main tasks are to add Soroban as a compilation target and on-boarding production projects to Solang.
```
Comment (yacovm) [inline project-reports/2024/2024-annual-Hyperledger-Solang.md:17]: Can we have a breakdown of each maintainer to its organizational association?

File: project-reports/2024/2024-annual-Hyperledger-Solang.md (hunk 1 above)
Comment (tkuhrt) [inline project-reports/2024/2024-annual-Hyperledger-Solang.md]:
```suggestion
Prior goals for Solang mainly involved making Solang more attractive to Solidity developers as stated in the 2023 [Q4 plans](https://github.com/hyperledger/toc/blob/gh-pages/project-reports/2023/2023-Q4-Hyperledger-Solang.md#current-plans)
```

File: project-reports/2024/2024-annual-Hyperledger-Solang.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+grand_parent: Hyperledger TOC
+nav_exclude: true
+---
+
+# Project Health
+
+The 2024 forecast for Solang is showing growth despite the number of pull requests and reported issues going down, Here are the three reasons why:
+1- Stellar organization decided to fund Solang long term, to add Soroban as a compilation target.
+
+2- Work is ongoing in Hyperledger Solang Playground, a web IDE that will enable easy onboarding to the Solang ecosystem. Funding is provided by the Web3 Foundation.
```
Comment (yacovm) [inline project-reports/2024/2024-annual-Hyperledger-Solang.md]: add link to https://github.com/hyperledger-labs/solang-playground ?

File: project-reports/2024/2024-annual-Hyperledger-Solang.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+# Project Health
+
+The 2024 forecast for Solang is showing growth despite the number of pull requests and reported issues going down, Here are the three reasons why:
+1- Stellar organization decided to fund Solang long term, to add Soroban as a compilation target.
+
+2- Work is ongoing in Hyperledger Solang Playground, a web IDE that will enable easy onboarding to the Solang ecosystem. Funding is provided by the Web3 Foundation. 
+
+3- Hyperledger Solang-Aqd (a smart contract CLI tool) was developed in the 2023 mentorship, by Tarek Naser, and it will be integrated into Solang Playground.
```
Comment (yacovm) [inline project-reports/2024/2024-annual-Hyperledger-Solang.md]: add link to https://github.com/hyperledger/solang-aqd ?

File: project-reports/2024/2024-annual-Hyperledger-Solang.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+# Goals
+
+## Performance Against Prior Goals
+Prior goals for Solang mainly involved making Solang more attractive to Solidity developers as stated in the 2023 [Q4 plans](https://github.com/hyperledger/toc/blob/gh-pages/project-reports/2023/2023-Q4-Hyperledger-Solang.md#current-plans)
+
+There have been many contributions to that cause, but they involved Solang itself and not its sidecar tools.
+
+
+## Next Year's Goals
+The main goal is to on-board more users and production projects to Hyperledger Solang.
```
Comment (yacovm) [inline project-reports/2024/2024-annual-Hyperledger-Solang.md]: how is this to be achieved?

Comment (salaheldinsoliman) [reply project-reports/2024/2024-annual-Hyperledger-Solang.md]: I have added a line that says more work is to be directed towards tooling and advocacy

**Review by yacovm — APPROVED** [review]

File: project-reports/2024/2024-annual-Hyperledger-Solang.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+---
+layout: default
+title: Hypereledger Solang
+parent: 2024
+grand_parent: Hyperledger TOC
+nav_exclude: true
```
Comment (ryjones) [inline project-reports/2024/2024-annual-Hyperledger-Solang.md]: delete this line:

`nav_exclude: true`

**Review by yacovm — APPROVED** [review]

File: project-reports/2024/2024-annual-Hyperledger-Solang.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
+---
+layout: default
+title: Hypereledger Solang
```
Comment (seanyoung) [inline project-reports/2024/2024-annual-Hyperledger-Solang.md]: s/Hypereledger/Hyperledger

**Comment by yacovm** [comment]
This year, the state of Hyperledger Solang is still pretty good, as the project maintainers anticipate growth despite a decline in pull requests. 

A positive point is that I did see quite a few new contributions (of new contributors) in the last year.

Reasons to be optimistic are new funding from Stellar, contribution of of Solang Playground from Web3 Foundation, and the contribution of of Solang-Aqd, a smart contract CLI tool. 

There has been a diversification  maintainers due to the the addition of Salaheldin Soliman, as now the maintainers are from three different organizations totalling Solana, Parity, and Stellar. However I cannot see @salaheldinsoliman in the maintainers file. 

The main difficulty the maintainers report is adoption due to insufficient developer tooling, and they will try to address this by creating tutorials and user-friendly tools. They will also try to onboard more users to Solang. 

Another notable thing Solang lacks is technical writing and review of tutorials and explanations. 

I overall agree with the recommendation of the maintainers to have the project remain in incubation, and I would like to add that it can be good to elaborate on how they are going to achieve their goals for this year (onboarding new users).

**Review by yacovm — APPROVED** [review]

**Review by petermetz — APPROVED** [review]

**Comment by salaheldinsoliman** [comment]
@yacovm Thanks!
I will make a PR in Solang to add myself to the MAINTAINERS.txt file

**Comment by salaheldinsoliman** [comment]
@yacovm @ryjones Hello,
Is this ready to be merged?

**Comment by tkuhrt** [comment]
@salaheldinsoliman : We will be reviewing this at the TOC meeting this week on Thursday.

**Review by swcurran — APPROVED** [review]

**Comment by yacovm** [comment]
> @yacovm Thanks! I will make a PR in Solang to add myself to the MAINTAINERS.txt file

Still not here: https://github.com/hyperledger/solang/blob/main/MAINTAINERS.md

**Review by yacovm — APPROVED** [review]

**Comment by ryjones** [comment]
There is another maintainer that is not in a maintainers file. [https://github.com/hyperledger/solang-aqd](https://github.com/hyperledger/solang-aqd) has @tareknaser as a maintainer but there is no file.

File: project-reports/2024/2024-annual-Hyperledger-Solang.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+# Goals
+
+## Performance Against Prior Goals
+Prior goals for Solang mainly involved making Solang more attractive to Solidity developers as stated in the 2023 [Q4 plans](https://github.com/hyperledger/toc/blob/gh-pages/project-reports/2023/2023-Q4-Hyperledger-Solang.md#current-plans)
+
+There have been many contributions to that cause, but they involved Solang itself and not its sidecar tools.
+
+
+## Next Year's Goals
+The main goal is to on-board more users and production projects to Hyperledger Solang. More work will be put into Solang Playground and Solang-Aqd to make it easier for Solidity developers to start using Solang, as well as tutorials to encourage production projects to use Solang.
```
Comment (tkuhrt) [inline project-reports/2024/2024-annual-Hyperledger-Solang.md:36]: One of the suggestions from the TOC is to work on increasing maintainers as well as commit diversity.

**Review by VRamakrishna — APPROVED** [review]

**Review by denyeart — APPROVED** [review]

**Pull request merged**